diff --git a/.gitignore b/.gitignore
index f2f889294..c3297526f 100644
--- a/.gitignore
+++ b/.gitignore
@@ -29,6 +29,7 @@ TAGS
 # ignore default build directory, and directories from test-meson-builds.sh
 build
 build-*
-# ignore other build directory patterns
 *-gcc*
 *-clang*
+/build/
+/x86_64-native-linuxapp-gcc/
diff --git a/config/common_base b/config/common_base
index c7d5c7321..286250b42 100644
--- a/config/common_base
+++ b/config/common_base
@@ -49,11 +49,6 @@ CONFIG_RTE_FORCE_INTRINSICS=n
 #
 CONFIG_RTE_ARCH_STRICT_ALIGN=n
 
-#
-# Enable link time optimization
-#
-CONFIG_RTE_ENABLE_LTO=n
-
 #
 # Compile to share library
 #
@@ -64,6 +59,11 @@ CONFIG_RTE_BUILD_SHARED_LIB=n
 #
 CONFIG_RTE_NEXT_ABI=y
 
+#
+# Major ABI to overwrite library specific LIBABIVER
+#
+CONFIG_RTE_MAJOR_ABI=
+
 #
 # Machine's cache line size
 #
@@ -99,10 +99,10 @@ CONFIG_RTE_MAX_MEMZONE=2560
 CONFIG_RTE_MAX_TAILQ=32
 CONFIG_RTE_ENABLE_ASSERT=n
 CONFIG_RTE_LOG_DP_LEVEL=RTE_LOG_INFO
-CONFIG_RTE_ENABLE_TRACE_FP=n
 CONFIG_RTE_LOG_HISTORY=256
 CONFIG_RTE_BACKTRACE=y
 CONFIG_RTE_LIBEAL_USE_HPET=n
+CONFIG_RTE_EAL_ALLOW_INV_SOCKET_ID=n
 CONFIG_RTE_EAL_ALWAYS_PANIC_ON_ERROR=n
 CONFIG_RTE_EAL_IGB_UIO=n
 CONFIG_RTE_EAL_VFIO=n
@@ -111,11 +111,6 @@ CONFIG_RTE_MAX_VFIO_CONTAINERS=64
 CONFIG_RTE_MALLOC_DEBUG=n
 CONFIG_RTE_EAL_NUMA_AWARE_HUGEPAGES=n
 CONFIG_RTE_USE_LIBBSD=n
-# Use WFE instructions to implement the rte_wait_for_equal_xxx APIs,
-# calling these APIs put the cores in low power state while waiting
-# for the memory address to become equal to the expected value.
-# This is supported only by aarch64.
-CONFIG_RTE_ARM_USE_WFE=n
 
 #
 # Recognize/ignore the AVX/AVX512 CPU flags for performance/power testing.
@@ -125,9 +120,6 @@ CONFIG_RTE_ARM_USE_WFE=n
 CONFIG_RTE_ENABLE_AVX=y
 CONFIG_RTE_ENABLE_AVX512=n
 
-# Use ARM LSE ATOMIC instructions
-CONFIG_RTE_ARM_FEATURE_ATOMICS=n
-
 # Default driver path (or "" to disable)
 CONFIG_RTE_EAL_PMD_PATH=""
 
@@ -225,11 +217,12 @@ CONFIG_RTE_LIBRTE_BNXT_PMD=y
 # Compile burst-oriented Chelsio Terminator (CXGBE) PMD
 #
 CONFIG_RTE_LIBRTE_CXGBE_PMD=y
-
-#
-# Compile burst-oriented NXP PFE PMD driver
-#
-CONFIG_RTE_LIBRTE_PFE_PMD=n
+CONFIG_RTE_LIBRTE_CXGBE_DEBUG=n
+CONFIG_RTE_LIBRTE_CXGBE_DEBUG_REG=n
+CONFIG_RTE_LIBRTE_CXGBE_DEBUG_MBOX=n
+CONFIG_RTE_LIBRTE_CXGBE_DEBUG_TX=n
+CONFIG_RTE_LIBRTE_CXGBE_DEBUG_RX=n
+CONFIG_RTE_LIBRTE_CXGBE_TPUT=y
 
 # NXP DPAA Bus
 CONFIG_RTE_LIBRTE_DPAA_BUS=n
@@ -283,21 +276,6 @@ CONFIG_RTE_LIBRTE_E1000_DEBUG_TX=n
 CONFIG_RTE_LIBRTE_E1000_DEBUG_TX_FREE=n
 CONFIG_RTE_LIBRTE_E1000_PF_DISABLE_STRIP_CRC=n
 
-#
-# Compile burst-oriented HINIC PMD driver
-#
-CONFIG_RTE_LIBRTE_HINIC_PMD=n
-
-#
-# Compile burst-oriented HNS3 PMD driver
-#
-CONFIG_RTE_LIBRTE_HNS3_PMD=n
-
-#
-# Compile Pensando IONIC PMD driver
-#
-CONFIG_RTE_LIBRTE_IONIC_PMD=y
-
 #
 # Compile burst-oriented IXGBE PMD driver
 #
@@ -306,6 +284,7 @@ CONFIG_RTE_LIBRTE_IXGBE_DEBUG_RX=n
 CONFIG_RTE_LIBRTE_IXGBE_DEBUG_TX=n
 CONFIG_RTE_LIBRTE_IXGBE_DEBUG_TX_FREE=n
 CONFIG_RTE_LIBRTE_IXGBE_PF_DISABLE_STRIP_CRC=n
+CONFIG_RTE_IXGBE_INC_VECTOR=y
 CONFIG_RTE_LIBRTE_IXGBE_BYPASS=n
 
 #
@@ -338,46 +317,35 @@ CONFIG_RTE_LIBRTE_ICE_PMD=y
 CONFIG_RTE_LIBRTE_ICE_DEBUG_RX=n
 CONFIG_RTE_LIBRTE_ICE_DEBUG_TX=n
 CONFIG_RTE_LIBRTE_ICE_DEBUG_TX_FREE=n
+CONFIG_RTE_LIBRTE_ICE_RX_ALLOW_BULK_ALLOC=y
 CONFIG_RTE_LIBRTE_ICE_16BYTE_RX_DESC=n
 
 # Compile burst-oriented IAVF PMD driver
 #
 CONFIG_RTE_LIBRTE_IAVF_PMD=y
+CONFIG_RTE_LIBRTE_IAVF_INC_VECTOR=y
 CONFIG_RTE_LIBRTE_IAVF_DEBUG_TX=n
 CONFIG_RTE_LIBRTE_IAVF_DEBUG_TX_FREE=n
 CONFIG_RTE_LIBRTE_IAVF_DEBUG_RX=n
-CONFIG_RTE_LIBRTE_IAVF_DEBUG_DUMP_DESC=n
 CONFIG_RTE_LIBRTE_IAVF_16BYTE_RX_DESC=n
 #
 # Compile burst-oriented IPN3KE PMD driver
 #
-CONFIG_RTE_LIBRTE_IPN3KE_PMD=n
-
-#
-# Compile burst-oriented IGC PMD drivers
-#
-CONFIG_RTE_LIBRTE_IGC_PMD=y
-CONFIG_RTE_LIBRTE_IGC_DEBUG_RX=n
-CONFIG_RTE_LIBRTE_IGC_DEBUG_TX=n
+CONFIG_RTE_LIBRTE_IPN3KE_PMD=y
 
 #
 # Compile burst-oriented Mellanox ConnectX-3 (MLX4) PMD
 #
-CONFIG_RTE_LIBRTE_MLX4_PMD=n
+CONFIG_RTE_LIBRTE_MLX4_PMD=y
 CONFIG_RTE_LIBRTE_MLX4_DEBUG=n
 
 #
 # Compile burst-oriented Mellanox ConnectX-4, ConnectX-5,
-# ConnectX-6 & BlueField (MLX5) PMD
+# ConnectX-6 & Bluefield (MLX5) PMD
 #
-CONFIG_RTE_LIBRTE_MLX5_PMD=n
+CONFIG_RTE_LIBRTE_MLX5_PMD=y
 CONFIG_RTE_LIBRTE_MLX5_DEBUG=n
 
-#
-# Compile vdpa-oriented Mellanox ConnectX-6 & BlueField (MLX5) PMD
-#
-CONFIG_RTE_LIBRTE_MLX5_VDPA_PMD=n
-
 # Linking method for mlx4/5 dependency on ibverbs and related libraries
 # Default linking is dynamic by linker.
 # Other options are: dynamic by dlopen at run-time, or statically embedded.
@@ -437,11 +405,6 @@ CONFIG_RTE_LIBRTE_LIO_DEBUG_REGS=n
 #
 CONFIG_RTE_LIBRTE_OCTEONTX_PMD=y
 
-#
-# Compile burst-oriented Marvell OCTEON TX2 network PMD driver
-#
-CONFIG_RTE_LIBRTE_OCTEONTX2_PMD=y
-
 #
 # Compile WRS accelerated virtual port (AVP) guest PMD driver
 #
@@ -481,11 +444,6 @@ CONFIG_RTE_LIBRTE_PMD_AF_PACKET=n
 #
 CONFIG_RTE_LIBRTE_PMD_AF_XDP=n
 
-#
-# Compile Memory Interface PMD driver (Linux only)
-#
-CONFIG_RTE_LIBRTE_PMD_MEMIF=n
-
 #
 # Compile link bonding PMD library
 #
@@ -524,7 +482,7 @@ CONFIG_RTE_LIBRTE_NETVSC_DEBUG_DUMP=n
 #
 # Compile virtual device driver for NetVSC on Hyper-V/Azure
 #
-CONFIG_RTE_LIBRTE_VDEV_NETVSC_PMD=n
+CONFIG_RTE_LIBRTE_VDEV_NETVSC_PMD=
 
 #
 # Compile null PMD
@@ -563,11 +521,8 @@ CONFIG_RTE_PMD_PACKET_PREFETCH=y
 # EXPERIMENTAL: API may change without prior notice
 #
 CONFIG_RTE_LIBRTE_BBDEV=y
-CONFIG_RTE_LIBRTE_BBDEV_DEBUG=n
 CONFIG_RTE_BBDEV_MAX_DEVS=128
 CONFIG_RTE_BBDEV_OFFLOAD_COST=y
-CONFIG_RTE_BBDEV_SDK_AVX2=n
-CONFIG_RTE_BBDEV_SDK_AVX512=n
 
 #
 # Compile PMD for NULL bbdev device
@@ -577,17 +532,7 @@ CONFIG_RTE_LIBRTE_PMD_BBDEV_NULL=y
 #
 # Compile PMD for turbo software bbdev device
 #
-CONFIG_RTE_LIBRTE_PMD_BBDEV_TURBO_SW=y
-
-#
-# Compile PMD for Intel FPGA LTE FEC bbdev device
-#
-CONFIG_RTE_LIBRTE_PMD_BBDEV_FPGA_LTE_FEC=y
-
-#
-# Compile PMD for Intel FPGA 5GNR FEC bbdev device
-#
-CONFIG_RTE_LIBRTE_PMD_BBDEV_FPGA_5GNR_FEC=y
+CONFIG_RTE_LIBRTE_PMD_BBDEV_TURBO_SW=n
 
 #
 # Compile generic crypto device library
@@ -623,11 +568,6 @@ CONFIG_RTE_LIBRTE_DPAA_MAX_CRYPTODEV=4
 #
 CONFIG_RTE_LIBRTE_PMD_OCTEONTX_CRYPTO=y
 
-#
-# Compile PMD for Marvell OCTEON TX2 crypto device
-#
-CONFIG_RTE_LIBRTE_PMD_OCTEONTX2_CRYPTO=y
-
 #
 # Compile PMD for QuickAssist based devices - see docs for details
 #
@@ -699,11 +639,6 @@ CONFIG_RTE_LIBRTE_PMD_CCP=n
 #
 CONFIG_RTE_LIBRTE_PMD_MVSAM_CRYPTO=n
 
-#
-# Compile PMD for NITROX crypto device
-#
-CONFIG_RTE_LIBRTE_PMD_NITROX=y
-
 #
 # Compile generic security library
 #
@@ -768,11 +703,6 @@ CONFIG_RTE_LIBRTE_PMD_DSW_EVENTDEV=y
 #
 CONFIG_RTE_LIBRTE_PMD_OCTEONTX_SSOVF=y
 
-#
-# Compile PMD for octeontx2 sso event device
-#
-CONFIG_RTE_LIBRTE_PMD_OCTEONTX2_EVENTDEV=y
-
 #
 # Compile PMD for OPDL event device
 #
@@ -811,26 +741,6 @@ CONFIG_RTE_LIBRTE_PMD_DPAA2_QDMA_RAWDEV=n
 #
 CONFIG_RTE_LIBRTE_PMD_IFPGA_RAWDEV=n
 
-#
-# Compile PMD for Intel IOAT raw device
-#
-CONFIG_RTE_LIBRTE_PMD_IOAT_RAWDEV=y
-
-#
-# Compile PMD for octeontx2 DMA raw device
-#
-CONFIG_RTE_LIBRTE_PMD_OCTEONTX2_DMA_RAWDEV=y
-
-#
-# Compile PMD for octeontx2 EP raw device
-#
-CONFIG_RTE_LIBRTE_PMD_OCTEONTX2_EP_RAWDEV=y
-
-#
-# Compile PMD for NTB raw device
-#
-CONFIG_RTE_LIBRTE_PMD_NTB_RAWDEV=y
-
 #
 # Compile librte_ring
 #
@@ -861,11 +771,6 @@ CONFIG_RTE_DRIVER_MEMPOOL_STACK=y
 #
 CONFIG_RTE_LIBRTE_OCTEONTX_MEMPOOL=y
 
-#
-# Compile PMD for octeontx2 npa mempool device
-#
-CONFIG_RTE_LIBRTE_OCTEONTX2_MEMPOOL=y
-
 #
 # Compile librte_mbuf
 #
@@ -929,7 +834,7 @@ CONFIG_RTE_LIBRTE_BITRATE=y
 CONFIG_RTE_LIBRTE_LATENCY_STATS=y
 
 #
-# Compile librte_telemetry legacy support
+# Compile librte_telemetry
 #
 CONFIG_RTE_LIBRTE_TELEMETRY=n
 
@@ -939,17 +844,6 @@ CONFIG_RTE_LIBRTE_TELEMETRY=n
 CONFIG_RTE_LIBRTE_RCU=y
 CONFIG_RTE_LIBRTE_RCU_DEBUG=n
 
-#
-# Compile librte_rib
-#
-CONFIG_RTE_LIBRTE_RIB=y
-
-#
-# Compile librte_fib
-#
-CONFIG_RTE_LIBRTE_FIB=y
-CONFIG_RTE_LIBRTE_FIB_DEBUG=n
-
 #
 # Compile librte_lpm
 #
@@ -979,7 +873,7 @@ CONFIG_RTE_LIBRTE_NET=y
 #
 CONFIG_RTE_LIBRTE_IP_FRAG=y
 CONFIG_RTE_LIBRTE_IP_FRAG_DEBUG=n
-CONFIG_RTE_LIBRTE_IP_FRAG_MAX_FRAG=4
+CONFIG_RTE_LIBRTE_IP_FRAG_MAX_FRAG=64
 CONFIG_RTE_LIBRTE_IP_FRAG_TBL_STAT=n
 
 #
@@ -1048,6 +942,7 @@ CONFIG_RTE_PIPELINE_STATS_COLLECT=n
 CONFIG_RTE_LIBRTE_KNI=n
 CONFIG_RTE_LIBRTE_PMD_KNI=n
 CONFIG_RTE_KNI_KMOD=n
+CONFIG_RTE_KNI_KMOD_ETHTOOL=n
 CONFIG_RTE_KNI_PREEMPT_DEFAULT=y
 
 #
@@ -1087,18 +982,6 @@ CONFIG_RTE_LIBRTE_BPF_ELF=n
 #
 CONFIG_RTE_LIBRTE_IPSEC=y
 
-#
-# Compile librte_graph
-#
-CONFIG_RTE_LIBRTE_GRAPH=y
-CONFIG_RTE_GRAPH_BURST_SIZE=256
-CONFIG_RTE_LIBRTE_GRAPH_STATS=y
-
-#
-# Compile librte_node
-#
-CONFIG_RTE_LIBRTE_NODE=y
-
 #
 # Compile the test application
 #
diff --git a/config/meson.build b/config/meson.build
index 43ab11310..2d2855dea 100644
--- a/config/meson.build
+++ b/config/meson.build
@@ -204,7 +204,7 @@ warning_flags = [
 	'-Wwrite-strings',
 
 	# globally disabled warnings
-	'-Wno-address-of-packed-member',
+	#'-Wno-address-of-packed-member',
 	'-Wno-packed-not-aligned',
 	'-Wno-missing-field-initializers'
 ]
diff --git a/mk/rte.app.mk b/mk/rte.app.mk
index 0ce8cf541..86990fe0a 100644
--- a/mk/rte.app.mk
+++ b/mk/rte.app.mk
@@ -422,6 +422,11 @@ O_TO_EXE = $(LD) -o $@ $(OBJS-y) \
 	$(LDLIBS) $(LDFLAGS) $(LDFLAGS_$(@)) $(EXTRA_LDFLAGS) \
 	$(MAPFLAGS)
 endif
+
+# get DPDK to divulge the correct linker flags (borrowed from mTCP).
+$(shell if [ ! -d ${RTE_SDK}/${RTE_TARGET}/lib ]; then mkdir -p ${RTE_SDK}/${RTE_TARGET}/lib; fi)
+$(shell echo $(call linkerprefix,$(LDLIBS)) > ${RTE_SDK}/${RTE_TARGET}/lib/ldflags.txt)
+
 O_TO_EXE_STR = $(subst ','\'',$(O_TO_EXE)) #'# fix syntax highlight
 O_TO_EXE_DISP = $(if $(V),"$(O_TO_EXE_STR)","  LD $(@)")
 O_TO_EXE_CMD = "cmd_$@ = $(O_TO_EXE_STR)"
diff --git a/mk/rte.cpuflags.mk b/mk/rte.cpuflags.mk
index fa8753531..28578bde9 100644
--- a/mk/rte.cpuflags.mk
+++ b/mk/rte.cpuflags.mk
@@ -122,3 +122,7 @@ space:= $(empty) $(empty)
 CPUFLAGSTMP1 := $(addprefix RTE_CPUFLAG_,$(CPUFLAGS))
 CPUFLAGSTMP2 := $(subst $(space),$(comma),$(CPUFLAGSTMP1))
 CPUFLAGS_LIST := -DRTE_COMPILE_TIME_CPUFLAGS=$(CPUFLAGSTMP2)
+
+# get DPDK to divulge the correct compiler flags (borrowed from mTCP).
+$(shell if [ ! -d ${RTE_SDK}/${RTE_TARGET}/include ]; then mkdir -p ${RTE_SDK}/${RTE_TARGET}/include; fi)
+$(shell echo ${MACHINE_CFLAGS} > ${RTE_SDK}/${RTE_TARGET}/include/cflags.txt)
diff --git a/usertools/dpdk-setup.sh b/usertools/dpdk-setup.sh
index e5bbe9fee..62cfd1498 100755
--- a/usertools/dpdk-setup.sh
+++ b/usertools/dpdk-setup.sh
@@ -76,7 +76,7 @@ setup_target()
 		fi
 	fi
 	if [ "$QUIT" == "0" ] ; then
-		make install T=${RTE_TARGET}
+		make -j install T=${RTE_TARGET}
 	fi
 	echo "------------------------------------------------------------------------------"
 	echo " RTE_TARGET exported as $RTE_TARGET"
diff --git a/drivers/net/i40e/i40e_ethdev.h b/drivers/net/i40e/i40e_ethdev.h
index e5d0ce5..2b08f6e 100644
--- a/drivers/net/i40e/i40e_ethdev.h
+++ b/drivers/net/i40e/i40e_ethdev.h
@@ -180,7 +180,7 @@ enum i40e_flxpld_layer_idx {
 /* Default queue interrupt throttling time in microseconds */
 #define I40E_ITR_INDEX_DEFAULT          0
 #define I40E_ITR_INDEX_NONE             3
-#define I40E_QUEUE_ITR_INTERVAL_DEFAULT 32 /* 32 us */
+#define I40E_QUEUE_ITR_INTERVAL_DEFAULT 0 /* 32 us */
 #define I40E_QUEUE_ITR_INTERVAL_MAX     8160 /* 8160 us */
 #define I40E_VF_QUEUE_ITR_INTERVAL_DEFAULT 32 /* 32 us */
 /* Special FW support this floating VEB feature */
